docs: new blog post for release candidate (#3209)
keptn/lifecycle-toolkit: docs/blog/posts/stable-release-announcement.md

---
date: 2024-03-10
authors: [agardnerIT]
description: >
  In this blog post you will learn how to configure Keptn via the new Helm umbrella chart.
comments: true
---

# Announcing the Official Release Candidate for Keptn v2

The Keptn project is proud to announce a release candidate for what will become Keptn v2.
We are sharing this release candidate
so that the community and end users can participate in the testing
to ensure that the actual release is as robust as possible.
We anticipate (as long as no release blockers are identified)
that Keptn v2 will be released approximately 1 month after this announcement.

## What Problem Does Keptn Solve?

Deploying software using Kubernetes is, on the surface, easy.
Just use kubectl or a GitOps solution like ArgoCD to deploy a YAML file and that’s it, right?

> Well, no. Not normally.

Reviewer: ```suggestion
Well, no.
Not normally.
```

There is a lot more to ensuring a healthy deployment.
The pod(s) may be running, but that doesn’t automatically mean that the application is healthy.

> This is the problem domain that Keptn acts upon. By using Keptn, in combination with your standard

Reviewer: ```suggestion
This is the problem domain that Keptn acts upon.
By using Keptn, in combination with your standard
```

deployment tooling or practices, you can move from “I guess it’s OK” to “I know it’s OK”.
Keptn allows you to wrap governance and automated checks around the deployment process to ensure that
the end-to-end process of deploying is healthy and your application is meeting the SLOs you’ve defined.

## What’s New in Keptn v2?

User feedback to the Keptn project has been clear, and we’ve listened.

> Keptn v2 will bring 3 key new features:

Reviewer: ```suggestion
Keptn v2 will bring the following new features:
```

> 1.	Non-Blocking Tasks
> 2.	A new “promotion” stage

Reviewer: ```suggestion
- Non-Blocking Tasks
- A new “promotion” stage
```

Often, this is the behaviour you want – if a downstream dependency is unavailable or unhealthy,
you probably don’t want to complete the deployment.
However, for new Keptn users, this behaviour can appear drastic and cause deployments to be
“pending” without an obvious cause.
In Keptn v2 this blocking behaviour for pre tasks and evaluations is now configurable.
When creating the KeptnConfig resource, set `spec.blockDeployment: [true|false]`.
The default behaviour is for Keptn to block deployments (i.e. `spec.blockDeployment: true`)

### The Promotion Stage

Keptn v2 introduces a new “promotion” stage.
Keptn is commonly used alongside [GitOps practices](https://opengitops.dev/) and thus,
users want to have a dedicated way to promote an application to the next stage in my environment.
The new stage is disabled by default and can be controlled via the Helm flag:
`lifecycleOperator.promotionTasksEnabled: [true|false]`.
The upcoming stable release will be shipped with this feature enabled out of the box.

Further information can be found in
[the official documentation](https://keptn.sh/stable/docs/guides/multi-stage-application-delivery/).

> ## Try it out!

Reviewer: ```suggestion
## Try it out
```

We are sharing this release candidate
so that the community and end users can participate in the testing
to ensure that the actual release is as robust as possible.
We anticipate (as long as no release blockers are identified)
that Keptn v2 will be released approximately 1 month after this announcement.

## What Problem Does Keptn Solve?

Deploying software using Kubernetes is, on the surface, easy.
Just use kubectl or a GitOps solution like ArgoCD to deploy a YAML file and that’s it, right?
Well, no.
Not normally.
There is a lot more to ensuring a healthy deployment.
The pod(s) may be running, but that doesn’t automatically mean that the application is healthy.
This is the problem domain that Keptn acts upon.
By using Keptn, in combination with your standard
deployment tooling or practices, you can move from “I guess it’s OK” to “I know it’s OK”.
Keptn allows you to wrap governance and automated checks around the deployment process to ensure that
the end-to-end process of deploying is healthy and your application is meeting the SLOs you’ve defined.

## What’s New in Keptn v2?

User feedback to the Keptn project has been clear, and we’ve listened.

> Keptn v2 will bring the following new features:
> - Non-Blocking Tasks

Reviewer: ```suggestion
Keptn v2 will bring the following new features:

- Non-Blocking Tasks
```

---
date: 2024-03-10
authors: [agardnerIT]
description: >

>   In this blog post, the release candidate for Keptn v2 is announced.

Reviewer 2: ```suggestion
  This blog post announces the release candidate for Keptn v2.
```
Active voice is generally better than passive.

>   In this blog post you will learn how to configure Keptn via the new Helm umbrella chart.

Reviewer 3: I guess the description is a copy paste error, no?

Reviewer 3: from Keptn umbrella charts blogpost

Reviewer: good catch, fixed

>   In this blog post you will learn how to configure Keptn via the new Helm umbrella chart.

Reviewer: ```suggestion
  In this blog post, the release candidate for Keptn v2 is announced.
```

comments: true
---

# Announcing the Official Release Candidate for Keptn v2

The Keptn project is proud to announce a release candidate for what will become Keptn v2.
We are sharing this release candidate
so that the community and end users can participate in the testing
to ensure that the actual release is as robust as possible.
We anticipate (as long as no release blockers are identified)

> that Keptn v2 will be released approximately 1 month after this announcement.

Reviewer 3: do we really want to make promises?

Author: yes, that's the ETA of the Stable Release

Reviewer: this reads more like an estimation rather than a promise

Reviewer 2: Yes, that also concerns me.  On the other hand, we kind of need to tell people how long they have to test this out and have their comments matter.

## What Problem Does Keptn Solve?

Deploying software using Kubernetes is, on the surface, easy.
Just use kubectl or a GitOps solution like ArgoCD to deploy a YAML file and that’s it, right?
Well, no.
Not normally.
There is a lot more to ensuring a healthy deployment.
The pod(s) may be running, but that doesn’t automatically mean that the application is healthy.
This is the problem domain that Keptn acts upon.
By using Keptn, in combination with your standard
deployment tooling or practices, you can move from “I guess it’s OK” to “I know it’s OK”.
Keptn allows you to wrap governance and automated checks around the deployment process to ensure that
the end-to-end process of deploying is healthy and your application is meeting the SLOs you’ve defined.

> ## What’s New in Keptn v2?

Reviewer: ```suggestion
## What’s New in Keptn v2-rc?
```

> User feedback to the Keptn project has been clear, and we’ve listened.

Reviewer 2: Is this sentence really necessary and does it add anything?

Reviewer 2: Okay, but maybe not as the topic sentence for this section.  I'm rewording and moving this up to the beginning of the blog.

> Keptn v2 will bring the following new features:

Reviewer: ```suggestion
Keptn v2-rc will bring the following new features:
```

- Non-Blocking Tasks

> - A new “promotion” stage

Reviewer 3: These are the features present in the latest release, but are these the biggest achievements/features of Keptn v2?

Reviewer: agreed, changing it to v2-rc will make it clearer - I adapted it

> ### Non-Blocking Tests

Reviewer 2: ```suggestion
### Non-blocking tasks and evaluations
```

> Keptn offers the ability to perform arbitrary tasks and SLO evaluations both before a deployment and after a deployment.

Reviewer 2: Is "arbitrary tasks" the terminology we want here?  What about "...ability to define tasks and SLO evaluations..."?

> By design, any pre-deployment task or SLO evaluation which fails will block the deployment.

Reviewer 2: ```suggestion
By design, any pre-deployment task or SLO evaluation that fails will block the deployment.
```

Often, this is the behaviour you want – if a downstream dependency is unavailable or unhealthy,
you probably don’t want to complete the deployment.
However, for new Keptn users, this behaviour can appear drastic and cause deployments to be

> “pending” without an obvious cause.

Reviewer 2: Isn't the real purpose of this feature to allow people to test tasks and evalutions without blocking a deployment because of a bug in their code?

Reviewer 2: Yes, that was what I meant -- but I like your phraseology better.

> In Keptn v2 this blocking behaviour for pre tasks and evaluations is now configurable.

Reviewer 2: ```suggestion
In Keptn v2 this blocking behaviour for pre tasks and evaluations can be temporarily disabled for the cluster
until you are sure that your tasks and evaluations
are performing appropriately.
```

> When creating the KeptnConfig resource, set `spec.blockDeployment: [true|false]`.

Reviewer 2: You don't have to be creating KeptnConfig -- it could already exist and you could just add this setting.

The default behaviour is for Keptn to block deployments (i.e. `spec.blockDeployment: true`)

### The Promotion Stage

> Keptn v2 introduces a new “promotion” stage.

Reviewer 2: ```suggestion
Keptn v2 introduces a new “promotion” stage
to support multi-stage application delivery.
```
We are using both terms for this feature so let's use them both in the topic sentence.

> Keptn is commonly used alongside [GitOps practices](https://opengitops.dev/) and thus,

Reviewer 2: Is "alongside" really the term we want?  I thought Keptn is frequently used as part of GitOps practices.

users want to have a dedicated way to promote an application to the next stage in my environment.
The new stage is disabled by default and can be controlled via the Helm flag:
`lifecycleOperator.promotionTasksEnabled: [true|false]`.
The upcoming stable release will be shipped with this feature enabled out of the box.

Further information can be found in

> [the official documentation](https://keptn.sh/stable/docs/guides/multi-stage-application-delivery/).

Reviewer 2: ```suggestion
The
[Multi-stage application delvery](https://keptn.sh/stable/docs/guides/multi-stage-application-delivery/)
guide.
```
Better to use the title of the page in an xref

## Try it out

> Now, you will see a true end-to-end picture of everything in the logical operation,

Reviewer 2: ```suggestion
Now, you can see a true end-to-end picture of everything in the logical operation,
```

potentially from “PR merged” all the way to “deployment complete”.
[Download Keptn v2 RC1 Now!](https://artifacthub.io/packages/helm/lifecycle-toolkit/keptn)

> [Keptn v2 Release Candidate 1](https://github.com/keptn/lifecycle-toolkit/releases) is available now on GitHub.

Reviewer 2: Do we want two different locations like this.  If so, I suggest an unordered list something like:
- From artifacthub: <url>
- From github: <url>

> Please provide any feedback via the #keptn Slack channel in the

Reviewer 2: This sounds like we only want to hear about problems.  Don't we also want to know if people tried it out and everything worked perfectly?

---

> date: 2024-03-10

Reviewer 4: i guess the date needs to be adjusted. the 10th already passed :D

Keptn allows you to wrap governance and automated checks around the deployment process to ensure that
the end-to-end process of deploying is healthy and your application is meeting the SLOs you’ve defined.

## What’s New in Keptn v2-rc?

User feedback to the Keptn project has been clear, and we’ve listened.
Keptn v2-rc will bring the following new features:

- Non-Blocking Tasks
- A new “promotion” stage

### Non-blocking tasks and evaluations

Keptn offers the ability to perform arbitrary tasks and SLO evaluations both before a deployment and after a deployment.
By design, any pre-deployment task or SLO evaluation that fails will block the deployment.
Often, this is the behaviour you want – if a downstream dependency is unavailable or unhealthy,
you probably don’t want to complete the deployment.
However, for new Keptn users, this behaviour can appear drastic and cause deployments to be
“pending” without an obvious cause.
In Keptn v2 this blocking behaviour for pre tasks and evaluations can be temporarily disabled for the cluster
until you are sure that your tasks and evaluations
are performing appropriately.

> When creating the KeptnConfig resource, set `spec.blockDeployment: [true|false]`.

Reviewer: ```suggestion
When creating the [KeptnConfig](../../docs/reference/crd-reference/config.md) resource, set `spec.blockDeployment: [true|false]`.
```

The default behaviour is for Keptn to block deployments (i.e. `spec.blockDeployment: true`)

### The Promotion Stage

Keptn v2 introduces a new “promotion” stage
to support multi-stage application delivery.
Keptn is commonly used alongside [GitOps practices](https://opengitops.dev/) and thus,
users want to have a dedicated way to promote an application to the next stage in my environment.
The new stage is disabled by default and can be controlled via the Helm flag:
`lifecycleOperator.promotionTasksEnabled: [true|false]`.
The upcoming stable release will be shipped with this feature enabled out of the box.

Further information can be found in
The

> [Multi-stage application delvery](https://keptn.sh/stable/docs/guides/multi-stage-application-delivery/)

Reviewer: ```suggestion
[Multi-stage application delivery](../../docs/guides/multi-stage-application-delivery.md)
```

guide.

## Try it out

Now, you can see a true end-to-end picture of everything in the logical operation,
potentially from “PR merged” all the way to “deployment complete”.
[Download Keptn v2 RC1 Now!](https://artifacthub.io/packages/helm/lifecycle-toolkit/keptn)

[Keptn v2 Release Candidate 1](https://github.com/keptn/lifecycle-toolkit/releases) is available now on GitHub.

> Please provide any feedback via the #keptn Slack channel in the

Reviewer 4: this is the link: https://cloud-native.slack.com/messages/keptn/

### The Promotion Stage

Keptn v2 introduces a new “promotion” phase
to support multi-stage application delivery.
Keptn is commonly used as part of [GitOps practices](https://opengitops.dev/) and thus,
users want a dedicated way to promote an application to the next stage in my environment.
In this release candidate,
the new stage is disabled by default and can be controlled via the Helm flag:
`lifecycleOperator.promotionTasksEnabled: [true|false]`.
We expect that the upcoming stable release will be shipped with this feature enabled out of the box.

Further information can be found in
The
[Multi-stage application delivery](../../docs/guides/multi-stage-application-delivery.md)
guide.

## Try it out

With this release of Keptn, you can see a true end-to-end picture of everything in the logical order,
potentially from “PR merged” all the way to “deployment complete”.

To try it out, download Keptn v2 RC1 now:

- From
[ArtifactHub](https://artifacthub.io/packages/helm/lifecycle-toolkit/keptn)
- From
[GitHub](https://github.com/keptn/lifecycle-toolkit/releases)

Please provide any feedback -- good and bad -- by posting in the

> [CNCF Slack workspace](https://communityinviter.com/apps/cloud-native/cncf)

Reviewer 4: why not use the link to our channel instead of the inviter?

Author: done in f578e53
